src/db.rs

```diff
@@ -148,13 +151,13 @@ impl DbInner {
             let current_memtable = guard.memtable();
             current_memtable.put(key, value);
             let table = current_memtable.table().clone();
-            let last_compacted = guard.state().core.last_compacted_wal_sst_id;
-            self.maybe_freeze_memtable(&mut guard, last_compacted);
+            let last_wal_id = guard.last_written_wal_id();
+            self.maybe_freeze_memtable(&mut guard, last_wal_id);
             table
         };
 
         if options.await_flush {
```

Comment:
> I'm guessing you care the memtable flush vs. durable difference because you're disabling the WAL? But even still, wouldn't you just pick up from the last successful memtable write when you resume? Is this feature for those that don't have a durable log (Kafka) in front, and can't easily pick up where they left off?

The reason I added the flushed vs durable notifications was totally orthogonal to the wal enabled stuff. Clients doing put/del always block until the write is durable of `await_flush` (agree we can consider renaming this - but lets discuss in a separate issue) is true - that's the only condition they should care about - is the put I just did durable or not - whether it's written to object store or not is meaningless to the client. I wanted a notification for when the memtable has been written to object store so that we can use it for clients waiting on backpressure. When the current memtable has been written, then the waiter should check whether the backpressure condition (max numbe r of imm tables) is cleared and resume their writes.

Reply:
> that's the only condition they should care about - is the put I just did durable or not - whether it's written to object store or not is meaningless to the client

This confuses me. How can the write be durable if it's not yet written to object storage?

> I wanted a notification for when the memtable has been written to object store so that we can use it for clients waiting on backpressure

Ahhh, gotcha. Ok, that makes sense.